src/graphql/operations/votes.ts

```diff
@@ -47,6 +51,28 @@ export default async function(parent, args, context, info) {
     return Promise.reject('request failed');
   }
 
+  if (requestedFields.space && votes.length > 0) {
+    const spaceIds = votes.map(vote => vote.space.id);
+    const query = `
+      SELECT id, settings FROM spaces
+      WHERE id IN (?) AND settings IS NOT NULL  
+    `;
+    try {
+      let spaces = await db.queryAsync(query, [spaceIds]);
+
+      spaces = Object.fromEntries(
+        spaces.map(space => [space.id, formatSpace(space.id, space.settings)])
+      );
+      votes = votes.map(vote => {
```

Comment:
if you want to mutate vote, then you don't need a map, you can just use forEach. 

If you want to use map, I would recommended that you do not mutate vote. 
```js
// pseudocode
votes.map(vote => {
  if(condition) return {...vote, space: spaces[vote.space.id]}
  return vote
}
```

Reply:
@midgerate Changed vote to not to mutate